munch/__init__.py

```diff
@@ -453,6 +453,17 @@ def munchify_cycles(obj):
         # Then finish munchifying lists and dicts inside obj (reusing munchified obj if cycles are encountered)
         return post_munchify(partial, obj)
 
+    def pre_munchify_cycles(obj):
```

Comment:
You can reuse `pre_munchify_cycles` in `munchify_cycles` (same code except the last line)

Reply:
refactored